platforms/software/software-diagnostics/src/integTest/groovy/org/gradle/api/tasks/diagnostics/DependencyInsightReportTaskIntegrationTest.groovy

```diff
@@ -343,111 +340,34 @@ org:leaf:1.0
 
         then:
-        outputContains """Dependency resolution failed because of conflict on the following module:
-   - org:leaf2 between versions 2.5, 1.5 and 1.0
-
-org:leaf2:2.5
-  Variant runtime:
-    | Attribute Name             | Provided     | Requested |
-    |----------------------------|--------------|-----------|
-    | org.gradle.category        | library      |           |
-    | org.gradle.libraryelements | jar          |           |
-    | org.gradle.status          | release      |           |
-    | org.gradle.usage           | java-runtime |           |
-   Selection reasons:
-      - By conflict resolution: between versions 2.5, 1.5 and 1.0
-
-org:leaf2:2.5
-\\--- org:toplevel3:1.0
-     \\--- conf
+        outputContains """org:leaf2:1.0 FAILED
+   Failures:
+      - Could not resolve org:leaf2:1.0.
+          - Conflict found for module 'org:leaf2': between versions 2.5, 1.5 and 1.0
 
-org:leaf2:1.0 -> 2.5
+org:leaf2:1.0 FAILED
 +--- org:middle1:1.0
 |    \\--- org:toplevel:1.0
 |         \\--- conf
 \\--- org:middle3:1.0
      \\--- org:toplevel4:1.0
           \\--- conf
 
-org:leaf2:1.5 -> 2.5
+org:leaf2:1.5 FAILED
+   Failures:
+      - Could not resolve org:leaf2:1.5. (already reported)
+
+org:leaf2:1.5 FAILED
 \\--- org:toplevel2:1.0
      \\--- conf
-"""
-    }
-
-    @Issue("https://github.com/gradle/gradle/issues/24356")
-    def "displays information about conflicting modules when failOnVersionConflict is used and afterResolve is used"() {
-        given:
-        mavenRepo.module("org", "leaf1").publish()
-        mavenRepo.module("org", "leaf2").publish()
-        mavenRepo.module("org", "leaf2", "1.5").publish()
-        mavenRepo.module("org", "leaf2", "2.5").publish()
-        mavenRepo.module("org", "leaf3").publish()
-        mavenRepo.module("org", "leaf4").publish()
-
-        mavenRepo.module("org", "middle1").dependsOnModules('leaf1', 'leaf2').publish()
-        mavenRepo.module("org", "middle2").dependsOnModules('leaf3', 'leaf4').publish()
-        mavenRepo.module("org", "middle3").dependsOnModules('leaf2').publish()
-
-        mavenRepo.module("org", "toplevel").dependsOnModules("middle1", "middle2").publish()
-
-        mavenRepo.module("org", "toplevel2").dependsOn("org", "leaf2", "1.5").publish()
-        mavenRepo.module("org", "toplevel3").dependsOn("org", "leaf2", "2.5").publish()
-
-        mavenRepo.module("org", "toplevel4").dependsOnModules("middle3").publish()
-
-        buildFile << """
-            repositories {
-                maven { url = "${mavenRepo.uri}" }
-            }
-
-            configurations {
-                conf {
-                    resolutionStrategy.failOnVersionConflict()
-                    incoming.afterResolve {
-                        // If executed, the below will cause the resolution failure on version conflict to be thrown, breaking dependency insight
-                        it.artifacts.artifacts
```

Comment:
This is technically a breaking change but IMO a very minor one. 

`it.artifacts.artifacts` will throw an exception if the graph has _any_ failures in it. 

Previously, we skipped running `afterResolve` if we had a version conflict failure (but ran it in every other case). 

So, now we always run afterResolve and as such this throws an exception when it didn't previously. 

The new behavior is correct IMO. I would be fine releasing this in a minor as it is more correct and can be considered a bug fix (afterResolve didnt run in certain circumstances). Note, the `ResolutionResult` can still be analyzed leniently in `afterResolve` here without throwing an exception, so there is value to executing `afterResolve` even during failure cases.

Reply:
Is there a difference now for handling version conflicts on lenient configurations? I am not sure, though might it be that before a lenient configuration that failed because of a version conflict would still resolve "normally", resolving the version conflict by the default strategy? And now, the files in the lenient configuration would not contain the artifacts from the dependency that had the version conflict?

I don't think that is necessarily bad, though I am not sure what the bigger effect is.

Reply:
Yeah without testing it myself I would assume you're right. 

But that's sort of the contract of lenient artifact selection -- where you don't get certain artifacts upon failure. I think this is an acceptable change